### PR TITLE
Job Improvement Project - Rune Knight (Part 2)

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #4708

* **Server Mode**: Renewal

* **Description of Pull Request**: 
  * kRO Changelog: http://ro.gnjoy.com/news/update/View.asp?seq=243&curpage=1

### CODE REVIEW

**Comment by ecdarreola** [comment]
Hundred Spear doesn't seem to do splash damage upon testing. Wind Cutter/Ignition Break working fine.

**Comment by samers1** [comment]
Hello, where's part 1? could you put the pull link?

**Comment by aleos89** [comment]
It was applied a long time ago in #2181.

File: src/map/battle.cpp [hunk 1]
```diff
@@ -2876,6 +2880,21 @@ static int battle_get_weapon_element(struct Damage* wd, struct block_list *src,
...
+					element = ELE_HOLY;
+			}
+			break;
+		case RK_DRAGONBREATH_WATER:
+			if (sc) {
+				if (sc->data[SC_LUXANIMA]) // Lux Anima has priority over Fighting Spirit
+					element = ELE_NEUTRAL;
+				else if (sc->data[SC_FIGHTINGSPIRIT])
+					element = ELE_GHOST;
+			}
```
Comment (Litro) [inline src/map/battle.cpp]: Missing break ? or fall through?

File: src/map/battle.cpp [hunk 2]
```diff
@@ -3440,6 +3459,9 @@ static void battle_calc_multi_attack(struct Damage* wd, struct block_list *src,s
 	}
 
 	switch (skill_id) {
+		case RK_WINDCUTTER:
+			if (sd && sd->weapontype1 == W_2HSWORD)
+				wd->div_ = 2;
```
Comment (Litro) [inline src/map/battle.cpp]: here too missing break?

**Comment by Angelic234** [comment]
1. use item Runstone_Lux(ID 22540)
2. attack a monster
3. map-server crash

**Comment by Ahmed-Chawqi** [comment]
Lux Anima Runestone gives a chance to autospell storm blast lvl 1
actually shouldn't works with skills.
just when use auto attack with weapon.

When you use lux anima stone then use a skill like Dragon breath, map-serv crash

**Comment by LordWhiplash** [comment]
In hundred spears change, HitCount is 5, this doesnt means that it will hit 5 times with the full damage on each hit? isn't it supposed to be -5 and divide full damage in five hits?

**Comment by ecdarreola** [comment]
Is it normal that RK Storm Blast specifically the one that procs from Lux Anima Runestone multiplies itself by the amount of units it hits?

**Comment by ecdarreola** [comment]
Storm blast processed by SC_LUXANIMA is abnormal. It multiplies by the number of hits or number of mobs on the area.

Not sure if this is the official behavior however when I test the Pertz Runestone, storm blast doesn't multiply its hit but only once.

Video: https://youtu.be/Q10zwyY2qPk

**Comment by LordWhiplash** [comment]
lux anima effect are missing this buffs:
+30% physical damage against all enemies sizes 
+30% critical damage
+30% melee physical damage
+30% long ranged physical damage

**Comment by LordWhiplash** [comment]
Physical damage against all enemies sizes is working great, thanks!
but the +30% critical damage, +30% melee physical damage, and +30% long ranged physical damage are still missing

**Comment by LordWhiplash** [comment]
Now is working fine, thanks!

**Comment by Masao87** [comment]
Do you have any ETA on this maybe? :)

**Pull request merged**